esp/bindings/http/platform/httpservice.cpp

```diff
@@ -450,6 +483,9 @@ int CEspHttpServer::processRequest()
         return 0;
     }
 
+    //need to ensure that the span is ended when out of scope Owend<ISpan> processingRequestSpan?
+    processingRequestSpan->End();
```

Comment:
This should be managed by a local class object - otherwise any exceptions thrown will not call this code.